### PR TITLE
Fixing Android issue being unstable when creating several screenshots

### PR DESCRIPTION
**Pull request by graux**
Fixing Android issue being unstable:
- Using convertFlutterSurfaceToImage from binding and using revertFlutterImage from integrationTestChannel.
- Moved revertFlutterImage within takeScreenshot to be reverted by the same code that takes the screenshot.
- Fixed warning when throwing exception in web.
- Updated handler function for setMethodCallHandler

### CODE REVIEW

File: lib/src/capture_screen.dart [hunk 1]
```diff
@@ -74,68 +67,42 @@ Future<List<int>> takeScreenshot(
...
-  // final rawBytes = await integrationTestChannel.invokeMethod<List<int>>(
-  //   'captureScreenshot',
-  //   <String, dynamic>{'name': screenshot},
-  // ) as List<int>?;
-  final rawBytes = await binding.takeScreenshot(screenshot);
-
+  await binding.pump();
+  binding.reportData ??= <String, dynamic>{};
+  binding.reportData!['screenshots'] ??= <dynamic>[];
+  integrationTestChannel.setMethodCallHandler((dynamic call) async {
```
Comment (graux) [inline lib/src/capture_screen.dart:79]: The original method used in master has two parameters.
This was causing an error, not sure if it is required by patrol, but this is the right signature with the `flutter_test` implementation.

Comment (jonasbadstuebner) [reply lib/src/capture_screen.dart:79]: To be honest, I don't know what method exactly you mean, but I will just resolve this conversation.
I will run my test pipeline after this and we will see if anything breaks with patrol.

**Comment by graux** [comment]
Hi  @jonasbadstuebner 
I have been trying to make my screenshots fork to work, no luck.
And I get the same error using yours.
When I do:
`flutter pub global activate -s git https://github.com/jonasbadstuebner/screenshots -x screenshots`

I get this error:
```
Failed to build screenshots:main:
Error: Couldn't resolve the package 'http' in 'package:http/http.dart'.
../../../../../.pub-cache/git/screenshots-75553f936ea3855da2f172390986d60e9a8ea3cb/lib/src/capture_screen.dart:6:8: Error: Not found: 'package:http/http.dart'
import 'package:http/http.dart' as http;
       ^
../../../../../.pub-cache/git/screenshots-75553f936ea3855da2f172390986d60e9a8ea3cb/lib/src/capture_screen.dart:34:23: Error: Method not found: 'Client'.
  final client = http.Client();
                      ^^^^^^
```
It seems using Flutter SDK >= 3.27 and http >= 1.3.0 causes this error.
Are you planning to release an updated version?

I've tried to update all the dependencies of other linked projects, but `tool_base` convert.dart has an error when using updated framework:
`class Utf8Decoder extends cnv.Utf8Decoder {`
And `cnv.Utf8Decoder` is a final class now and cannot be extended.

Are you planning to release a new version?
Thank you!

**Comment by jonasbadstuebner** [comment]
I pushed an update, please let me know if it works for you?

If I broke something on your end, feel free to drop a comment and let me know what you see.

**Comment by jonasbadstuebner** [comment]
FYI: I could reproduce the error you saw before and with my latest commit https://github.com/jonasbadstuebner/screenshots/commit/9f0539c9542aafb19bb2a311eae1490a17b44cc9 I don't see it anymore and `screenshots` is activated just fine.

This is my current `flutter --version` output:
```
Flutter 3.27.4 • channel stable • https://github.com/flutter/flutter.git
Framework • revision d8a9f9a52e (11 days ago) • 2025-01-31 16:07:18 -0500
Engine • revision 82bd5b7209
Tools • Dart 3.6.2 • DevTools 2.40.3
```

**Comment by jonasbadstuebner** [comment]
I bump the version if you confirm it working.

PS: I know this is not the cleanest way of doing things, but currently, it's only you and me using this repo, I think. :D So I focus on getting stuff fixed, not documented as well as it could be.

**Comment by graux** [comment]
Hi @jonasbadstuebner !
Thank you for looking into it.
The `http: any`  seems promising I will take a look this Friday.
I think I got part of it working adding a section:
```
dependency_overrides:
  http: ^1.3.0
```
I will let you know, thanks!

**Comment by jonasbadstuebner** [comment]
I think I removed the need for the dependency override.
The http dependency is updated in all repos and should be 1.3.0 now

**Comment by graux** [comment]
Hi @jonasbadstuebner.
I am getting this error both in Linux and macos when installing the tool:
```
> flutter pub global activate -s git https://github.com/jonasbadstuebner/screenshots -x screenshots
Downloading packages... . (1.2s)
+ archive 4.0.2
+ args 2.6.0
+ async 2.13.0
+ clock 1.1.2
+ collection 1.19.1
+ crypto 3.0.6
+ ffi 2.1.3
+ file 7.0.1
+ http 1.3.0
+ http_parser 4.1.2
+ intl 0.20.2
+ meta 1.16.0
+ path 1.9.1
+ platform 3.1.6
+ posix 6.0.1
+ process 5.0.3
+ resource_portable 3.1.0
+ screenshots 2.3.0 from git https://github.com/jonasbadstuebner/screenshots at 8e9ba8
+ source_span 1.10.1
+ string_scanner 1.4.1
+ term_glyph 1.2.2
+ tool_base 1.10.0 from git https://github.com/jonasbadstuebner/tool_base.git at f64a38
+ tool_mobile 1.10.0 from git https://github.com/jonasbadstuebner/tool_mobile.git at d7292f
+ typed_data 1.4.0
+ web 1.1.0
+ yaml 3.1.3
Building package executables... 
Failed to build screenshots:main:
../../../../../.pub-cache/git/tool_base-f64a38195f3ef56d803a1bbf8dfbfb88680f550e/lib/src/convert.dart:30:27: Error: The class 'Utf8Decoder' can't be extended outside of its library because it's a final class.
class Utf8Decoder extends cnv.Utf8Decoder {
                          ^
Failed to update packages.
```

**Comment by graux** [comment]
Hi again!
I've fixed and tested the issue in my branches. 
You have a pull request in the [tool_base](https://github.com/jonasbadstuebner/tool_base/pull/2) project.

Let me know if that works for you,
Cheers!

**Pull request merged**